torch_frame/data/mapper.py

```diff
@@ -341,15 +343,19 @@ def forward(
                     device)
             return feat_dict
 
-        tokenized_list: TextTokenizationOutputs = []
+        tokenized_outputs: TextTokenizationOutputs = []
         for i in tqdm(range(0, len(ser_list), self.batch_size),
                       desc="Tokenizing texts in mini-batch"):
             tokenized_batch: TextTokenizationOutputs = self.text_tokenizer(
                 ser_list[i:i + self.batch_size])
-            tokenized_list.extend(tokenized_batch)
-        for key in tokenized_list[0]:
+            if isinstance(tokenized_batch, Mapping):
+                tokenized_outputs.append(tokenized_batch)
+            else:
+                tokenized_outputs.extend(tokenized_batch)
+
+        for key in tokenized_outputs[0]:
             xs = []
-            for item in tokenized_list:
+            for item in tokenized_outputs:
```

Comment:
`for tokenized_batch in tokenized_outputs`
to align with your previous for-loop